fixes #66: clustering example (runtime & data-driven styling)

incanus opened the pull request:
Decided to keep this simpler than the scope in #66 in two ways: 

- Don't worry about toggling clusters on & off. While this is possible by using multiple layers and sources and toggling their visibility, it complicates things too much. 
- Removed the spin animation on the unclustered points until such functionality makes it into the SDK. 

Backlog: 

- [x] Wait for 3.5.0 final release? This relies on data-driven styling and is using [b1](https://github.com/mapbox/mapbox-gl-native/releases/tag/ios-v3.5.0-beta.1). 
- [x] Remove code signing-related changes from Xcode project. 
- [x] Add Swift version once we're happy with things. 
- [x] Make an image and update the actual examples site (https://github.com/mapbox/ios-sdk#to-add-a-new-example)

incanus commented:
GIF of current iteration: 

![anigif-1488235033](https://cloud.githubusercontent.com/assets/17722/23383228/4db18a96-fcfa-11e6-9853-d65399b8ecae.gif)

friedbunny commented:
You can target this against the `release-v3.5.0` branch here and we’ll merge that once v3.5.0 is released.

incanus commented:
Ok, I think this is ready for final review.

friedbunny commented on Examples/ObjectiveC/ClusteringExample.m:
```diff
@@ -0,0 +1,138 @@
+    circlesLayer.circleStrokeColor = [MGLStyleValue valueWithRawValue:[[UIColor whiteColor] colorWithAlphaComponent:0.75]];
+    circlesLayer.circleStrokeWidth = [MGLStyleValue valueWithRawValue:@2];
+    circlesLayer.circleColor = [MGLSourceStyleFunction functionWithInterpolationMode:MGLInterpolationModeInterval
+                                                                              stops:stops
```
With the current width limitations of the /ios-sdk site, we can’t use normal ObjC indentation because it ends up looking like this:

<img width="766" alt="screen shot 2017-03-01 at 11 21 30 am" src="https://cloud.githubusercontent.com/assets/1198851/23477072/aa61a65e-fe71-11e6-81c1-81eb12309078.png">

friedbunny replied:
I pushed https://github.com/mapbox/ios-sdk/compare/fb-clustering which you can use to preview how this branch will look as an example.

incanus replied:
Ok, check out 1007a37f25f8c14d210ed9fcd606cc34e33271d6, which I tested with Jekyll.

friedbunny commented on Examples/ObjectiveC/ClusteringExample.m:
```diff
@@ -0,0 +1,138 @@
+        CGFloat width = self.sprite.size.width;
+        CGRect rect = CGRectMake(point.x - width / 2, point.y - width / 2, width, width);
+
+        NSArray *clusters = [self.mapView visibleFeaturesInRect:rect inStyleLayersWithIdentifiers:[NSSet setWithObject:@"clusteredPorts"]];
```
Let’s add some comments about what we’re doing here, as `visibleFeaturesInRect: inStyleLayersWithIdentifiers:` is somewhat specialized.

incanus replied:
f97fec9f697f9580284d9d9bc1d0137c4604321c

incanus replied:
f97fec9f697f9580284d9d9bc1d0137c4604321c

friedbunny commented on Examples.xcodeproj/project.pbxproj:
```diff
@@ -122,6 +122,12 @@
+		DD5939E41E6639BA0009BEB2 /* ClusteringExample.swift in Sources */ = {isa = PBXBuildFile; fileRef = DD5939E31E6639BA0009BEB2 /* ClusteringExample.swift */; };
+		DDF943291E5DE63300545D0F /* ClusteringExample.m in Sources */ = {isa = PBXBuildFile; fileRef = DDF943281E5DE63300545D0F /* ClusteringExample.m */; };
+		DDF9432B1E5DEACC00545D0F /* ports.geojson in Resources */ = {isa = PBXBuildFile; fileRef = DDF9432A1E5DEACC00545D0F /* ports.geojson */; };
+		DDF943311E5E3AFB00545D0F /* sprite.png in Resources */ = {isa = PBXBuildFile; fileRef = DDF9432E1E5E3AFB00545D0F /* sprite.png */; };
```
We’ve been using .xcasset files for these — it keeps things cleaner. We should also pick a more descriptive name for this image.

incanus replied:
6ed0ad7119a33c413a00209c70df4884443ce534

friedbunny commented on Examples/ObjectiveC/ClusteringExample.m:
```diff
@@ -0,0 +1,138 @@
+    circlesLayer.predicate = [NSPredicate predicateWithFormat:@"%K == YES", @"cluster"];
+    [style addLayer:circlesLayer];
+
+    // Label cluster circles with a layer of text indicating feature count. Per text token convention, wrap the attibute in {}.
```
s/attitute/attribute

incanus replied:
b0dd6f5786ea252cd7efc119e7fd35915dd60996

friedbunny commented on Examples/Swift/ClusteringExample.swift:
```diff
@@ -0,0 +1,133 @@
+        circlesLayer.predicate = NSPredicate(format: "%K == YES", "cluster")
+        style.addLayer(circlesLayer)
+
+        // Label cluster circles with a layer of text indicating feature count. Per text token convention, wrap the attibute in {}.
```
Same typo.

incanus replied:
b0dd6f5786ea252cd7efc119e7fd35915dd60996

incanus commented:
Ok, cleaned up @friedbunny's suggestions and added the requisite Jekyll-side stuff in https://github.com/mapbox/ios-sdk/commit/30ffc78dbe620817e56ff4410c035813b2b06060 (in a new `release-v3.5.0` branch as we do in this repo.

friedbunny commented:
Awesome, thanks for the great example, @incanus. 🙇

incanus commented:
Cool, this is ready to go when [iOS 3.5.0](https://github.com/mapbox/mapbox-gl-native/milestone/48) is released!
